synapse/handlers/e2e_keys.py

```diff
@@ -241,24 +241,25 @@ async def query_devices(
             logger.debug(
                 "%d destinations to query devices for", len(remote_queries_not_in_cache)
             )
-            await make_deferred_yieldable(
-                delay_cancellation(
-                    defer.gatherResults(
-                        [
-                            run_in_background(
-                                self._query_devices_for_destination,
-                                results,
-                                cross_signing_keys,
-                                failures,
-                                destination,
-                                queries,
-                                timeout,
-                            )
-                            for destination, queries in remote_queries_not_in_cache.items()
-                        ],
-                        consumeErrors=True,
-                    ).addErrback(unwrapFirstError)
+
+            async def _query(
+                destination_queries: Tuple[str, Dict[str, Iterable[str]]]
+            ) -> None:
+                destination, queries = destination_queries
+                return await self._query_devices_for_destination(
+                    results,
+                    cross_signing_keys,
+                    failures,
+                    destination,
+                    queries,
+                    timeout,
                 )
+
+            await concurrently_execute(
+                _query,
+                remote_queries_not_in_cache.items(),
+                10,
+                delay_cancellation=True,
             )
```

Comment:
This is pretty much the same as what was here before but is now using a helper and limiting the number of concurrent queries?

(As far as I can tell, `concurrently_execute` is the same as `make_deferred_yieldable(defer.gatherResults([run_in_background(...)]))` and now with the `delay_cancellation` that you added?)

Reply:
yep, this part is just concurrency limiting